mtags/src/main/scala-2/scala/meta/internal/metals/MultilineStringFormattingProvider.scala

```diff
@@ -109,57 +130,111 @@ object MultilineStringFormattingProvider {
+      tokens: Tokens,
+      start: StartPosition,
+      end: EndPosition
+  ): PartialFunction[(Token, Int), Boolean] = {
+    case (token: Constant.String, index: Int) =>
+      isMultilineString(sourceText, token) && hasStripMarginSuffix(
+        index,
+        tokens
+      ) && inToken(start, end, token)
+  }
+  private def shouldFormatInterpolationString(
+      sourceText: String,
+      tokens: Tokens,
+      start: StartPosition
+  ): PartialFunction[(Token, Int), Boolean] = {
+    case (token: Interpolation.Start, index: Int)
+        if token.start < start.start => {
+      var endIndex = index + 1
+      while (!tokens(endIndex)
+          .isInstanceOf[Interpolation.End]) endIndex += 1
+      isMultilineString(sourceText, token) && hasStripMarginSuffix(
+        endIndex,
+        tokens
+      )
     }
-    shouldAddPipes
   }
 
   private def withToken(
       textId: TextDocumentIdentifier,
       sourceText: String,
-      range: Range,
-      newlineAdded: Boolean
-  )(fn: (String, meta.Position) => List[TextEdit]): List[TextEdit] = {
+      range: Range
+  )(
+      fn: (
+          StartPosition,
+          EndPosition,
+          String,
+          Option[Tokens]
+      ) => List[TextEdit]
+  ): List[TextEdit] = {
     val source = textId.getUri.toAbsolutePath
     if (source.exists) {
       val virtualFile = Input.VirtualFile(source.toString(), sourceText)
       val startPos = range.getStart.toMeta(virtualFile)
       val endPos = range.getEnd.toMeta(virtualFile)
-      if (pipeInScope(startPos, endPos, sourceText, newlineAdded)) {
-        val tokens =
-          virtualFile.tokenize.toOption
-        tokens
-          .filter(multilineStringInTokens(_, startPos, endPos, sourceText))
-          .map(_ => fn(sourceText, startPos))
-          .getOrElse(Nil)
-      } else Nil
+      fn(startPos, endPos, sourceText, virtualFile.tokenize.toOption)
     } else Nil
   }
 
+  private def doubleQuoteNotClosed(
+      splitLines: Array[String],
+      position: Position
+  ): Boolean = {
+    val lineBefore = splitLines(position.getLine - 1)
+    !getIndexOfLastQuote(lineBefore).exists(_._2)
+  }
+
+  private def fixStringNewline(
+      position: Position,
+      splitLines: Array[String]
+  ): List[TextEdit] = {
+    val previousLineNumber = position.getLine - 1
+    val previousLine = splitLines(previousLineNumber)
+    val previousLinePosition =
+      new Position(previousLineNumber, previousLine.length)
+    val textEditPrecedentLine = new TextEdit(
+      new Range(previousLinePosition, previousLinePosition),
+      "\"" + " " + "+"
+    )
+    val defaultIndent = previousLine.prefixLength(_ == ' ')
+    val indent =
+      if (previousLineNumber > 1 && !splitLines(previousLineNumber - 1).trim.lastOption
+          .contains('+'))
+        defaultIndent + 2
+      else defaultIndent
+    val interpolationString = {
+      val quoteIndex = getIndexOfLastQuote(previousLine).map(_._1).getOrElse(-1)
```

Comment:
Maybe this can be a bit more clear also?
```
 getIndexOfLastQuote(previousLine).map{
  case (lastQuoteIndex) => 
    if (quoteIndex >0 && previousLine(quoteIndex - 1) == 's') "s" else ""
}.getOrElse("")
```

Reply:
ok